test/jit/test_symbolic_shape_analysis.py

```diff
@@ -328,6 +326,39 @@ def forward(self, x, y):
             inps[2].setType(inps[2].type().with_sizes(args[1].size()))
             self.checkShapeAnalysis(out_size, mod.graph, assert_propagation=True)
 
+    def assert_shape_equal_scripted(self, script_fn, given_ins):
+        expected_res = script_fn(*given_ins)
+        g = script_fn.graph
+        graph_ins = list(g.inputs())
+        self.assertEqual(len(given_ins), len(graph_ins))
+        for inp, graph_in in zip(given_ins, graph_ins):
+            graph_in.setType(graph_in.type().with_sizes(inp.size()))
+
+        torch._C._jit_pass_propagate_shapes_on_graph(g)
+        if not isinstance(expected_res, tuple):
+            raise NotImplementedError("TODO: implement single output case")
+
+        tuple_construct_node = next(g.outputs()).node()
+        for exp, graph_out in zip(expected_res, list(tuple_construct_node.inputs())):
+            expected_size = list(exp.size())
+            symbolic_size = graph_out.type().symbolic_sizes()
+            self.assertEqual(symbolic_size, expected_size)
```

Comment:
can you replace with  https://github.com/pytorch/pytorch/blob/2fcd5808a315dff0bd12e82e055882102fcabd28/torch/testing/_internal/common_jit.py#L285 ?

Reply:
It errored out when I was using that instead of scripting the function.

Reply:
I'm suggesting

```
        expected_res = script_fn(*given_ins)
        g = script_fn.graph
        graph_ins = list(g.inputs())
        self.assertEqual(len(given_ins), len(graph_ins))
        for inp, graph_in in zip(given_ins, graph_ins):
            graph_in.setType(graph_in.type().with_sizes(inp.size()))

out_sizes= [out.size() for out in expected_res]
self.checkShapeAnalysis(out_sizes, graph_in, assert_propagation=True)
```